packages/host/app/services/sessions-service.ts

```diff
@@ -78,4 +87,85 @@ export default class SessionsService extends Service {
       }
     }
   }
+
+  async getRealmToken(
+    realmURL: URL,
+    skipCache?: boolean,
+  ): Promise<string | undefined> {
+    let tokenRefreshPeriod = 5 * 60; // 5 minutes
+    let rawJWT = this.realmURLToRawJWT.get(realmURL.href);
+
+    if (rawJWT && !skipCache) {
+      let claims = this.toJWT(rawJWT);
+      let expiration = claims.exp;
+      if (expiration - tokenRefreshPeriod > Date.now() / 1000) {
+        return rawJWT;
+      }
+    }
+
+    rawJWT = await this.createRealmSession(realmURL);
```

Comment:
We are doing a lot of the same things compared to [my PR](https://github.com/cardstack/boxel/pull/1014/files#diff-24d76ba72eb2961323a9fbeabd16ee1c9a1c195583e4b44ef94fb9d6aee3ea94R22), we should consolidate the logic, ideally reuse

Reply:
I think I can reuse your `RealmAuthClient` class.